### DISCUSSION TITLE
Temporal tutorial

### DISCUSSION THREAD

**Discussion by mcunningham1440** [pyg-team/pytorch_geometric#7997] [Q&A] [answered]
Is there a tutorial article or video on how to add a temporal dimension to graphs, and how that works with dataloaders? My understanding is that, for node_attr, this is to represent it as a tensor of size [n_nodes, n_timepoints, n_features], but how do you add a temporal dimension to edge_index and edge_attr?

**Comment by rusty1s** [comment, marked as answer]
There doesn't exist a dedicated tutorial for this on PyG side, but there exists an extension via PyTorch Geometric Temporal that you may found useful. We are also working on better internal temporal support in PyG itself, so please stay tuned.

Currently, the best way to do it in PyG is to treat it similarly as normal batching, in the sense that you include the temporal dimension into the node dimension, giving you feature matrices of size `[num_timepoints * num_nodes/num_edges, num_features]`.

**Reply by mcunningham1440** [reply]
Makes sense--thanks!
